### PR TITLE
[Feature] chunkflow enhancements

### PR DESCRIPTION
**Pull request by jingpengw**
- [x] add PyTorch backend for inference using GPUs. Can only use 1 device though. Multiple GPU support is not implemented.
- [x] use image in higher mip level to validate the downloaded image chunk.
- [x] use template matching to detect black boxes in the downsampled image.
- [x] support in device patch masking. Kisuk implemented the masking in gpu, we should be able to use this feature.
- [x] add inference tests.
- [ ] support image mask. Tommy will not mask out the folds and cracks since they would like to use it to evaluate the performance of alignment. I will do it here.
- [ ] support kempression/fpzip
- [x] quantize and downsample the affinitymap and upload to a new layer for visual inspection.
- [x] add log in the cloud. we can just create a log file for each task and upload it to cloud storage.
- [ ] accept the output as a dictionary and this will really enable this package workable with multitask convnet.
- [x] test igneous with multiple processes.
- [ ] make it possible to use cpu and gpu at the same time. the message in queue should only be locations, and all the rest should be handled in the client. In this case, I should split the code to chunkflow and make the command line tool strong to use different backends.

implemented, not going to merge soon:
- [x] async patch provider. This will help if we use gpu for inference. This seems not urgent for now since we can run multiple processes in each instance. https://github.com/seung-lab/chunkflow/pull/5

not going to do:
- [ ] use less IP addresses using kubernetes. We have enough ip addresses for GPU instances.
- [ ] add real time speed monitor. we can use google stack driver used in igneous. I have already added log in GCS for speed recording.

### CODE REVIEW

**Comment by jingpengw** [comment]
@william-silversmith this was used in minnie10 production run. Can we merge?
I have rebased to master branch.

File: igneous/task_execution.py [hunk 1]
```diff
@@ -25,21 +25,25 @@ def handler(signum, frame):
...
+def command(tag, m, np, interval, queue, server, qurl, loop):
   if not m:
     execute(tag, queue, server, qurl, loop)
     return 
 
   # if multiprocessing
-  proc = mp.cpu_count()
-  pool = mp.Pool(processes=proc)
-  print("Running %s threads of execution." % proc)
+  if np <= 0:
```
Comment (william-silversmith) [inline igneous/task_execution.py]: np ususally means numpy.

File: requirements.txt [hunk 2]
```diff
@@ -7,3 +7,4 @@ pytest>=3.3.1
 task-queue>=0.4.1
 -e git+https://github.com/seung-lab/tqdm.git#egg=tqdm
 oauth2client==3.0.0
+scikit-image>=0.14.1
```
Comment (william-silversmith) [inline requirements.txt]: Might be a good idea to have a separate requirements file for chunkflow only stuff. People have complained to me about the number of dependencies igneous has.

Comment (jingpengw) [reply requirements.txt]: I agree. igneous have too much dependency. I had some trouble dealing with package version conflicts which I am not using in chunkflow. 
Actually, I am working on separating chunkflow as standalone package.

Comment (jingpengw) [reply requirements.txt]: In that case, we can just close this PR?

Comment (william-silversmith) [reply requirements.txt]: If that's what you'd like to do.

**Pull request closed without merging**